src/modification.cpp

```diff
@@ -0,0 +1,79 @@
+/** \brief This modification file calculates the percentage of packets lost
+*
+* This file defines a function filter2 which takes two files as parameters
+* 
+* The first parameter is the input file which it reads
+* The second parameter is the output file where it writes the modified output
+*
+* Cristina Ruiz Martin
+* ARSLab - Carleton University
+*
+*/
+
+#include <iostream>
+#include <string.h>
+#include <stdio.h>
+
+using namespace std;
+
+void filter2(char input_file[], char output_file[]){
+    FILE * fp;
+    FILE * fp_out;
+	
+	fp = fopen(input_file,"r");
+    fp_out = fopen(output_file,"w");
+	
+    fseek(fp, 0, SEEK_END);
+    long size = ftell(fp);
+    fseek(fp, 0, SEEK_SET);
+    char *pData = new char[size+1];
+    fread(pData, sizeof(char),size,fp);
+    fclose(fp);
+    
+    /**
+    * Parsing the read data from csv file
+    */
+    char* pch;
+    pch = strtok(pData, ",");
+    float counter = 0.0, counter1 = 0.0, counter2 = 0.0, pkt_loss = 0.0, pkt_loss_percnt = 0.0;
+    int rounding = 0;
+	
+    while(pch != NULL){
+        /** 
+	* Incrementing the counter everytime string matches with "dataOut"
+	*/
+        if (strcmp(pch,"dataOut")==0){
+            counter++;
+	}
+	/** 
+	* Incrementing the counter1 everytime string matches with "ackReceivedOut"
+	*/
+        if (strcmp(pch,"ackReceivedOut")==0){
+            counter1++;
+	}
+	/** 
+	* Incrementing the counter2 everytime string matches with "packetSentOut"
+	*/
+        if (strcmp(pch,"packetSentOut")==0){
+            counter2++;
+        }
+        pch = strtok(NULL, ",");
+    }
+    
+    fprintf(fp_out,"\nThe total number of packets sent, dataOut = %f\n",counter);
+    fprintf(fp_out,"\nThe total number of acknowledgements received, ackReceivedOut = %f\n",counter1);
```

Comment:
Please reduce the number of characters for lines 64 & 65. You will have to continue the print statement to the next line.

Reply:
Done

Reply:
Could you do it again.

Reply:
https://stackoverflow.com/questions/1752079/in-c-can-a-long-printf-statement-be-broken-up-into-multiple-lines
check this link to divide into multiple lines

Reply:
It is done.